app/views/root/local_transaction.html.erb

```diff
@@ -58,3 +58,15 @@
     </div>
   </section>
 <% end %>
+
+<% if @postcode_error %>
+  <script type="text/javascript">
+      GOVUK.analytics.trackEvent(
+        "userAlerts:LocalTransaction",
+        "postcodeErrorShown:<%= @postcode_error %>",
+        {
+          label: "<%= t @location_error || 'Please enter a valid full UK postcode.' %>"
+        }
+      )
+  </script>
```

Comment:
There was a comment from @fofr on a previous version of this diff about avoiding dropping a script tag directly on the page.  Could we (do we have a way to) communicate with analytics from the server?  Even if we did would it be bad because this would a) mess up request/response time or b) not include useful client-side details?